Source/JavaScriptCore/runtime/WriteBarrier.h

```diff
@@ -352,3 +352,24 @@ class WriteBarrierStructureID {
 };
 
 } // namespace JSC
+
+namespace WTF {
+
+template<typename T> struct VectorTraits<JSC::WriteBarrier<T>> : public SimpleClassVectorTraits {
+    static_assert(std::is_trivially_destructible<JSC::WriteBarrier<T>>::value);
+    static const bool canInitializeWithMemset = true;
+    static const bool canCopyWithMemcpy = true;
+};
+
+template<> struct VectorTraits<JSC::WriteBarrier<JSC::Unknown>> : public SimpleClassVectorTraits {
+    static_assert(std::is_trivially_destructible<JSC::WriteBarrier<JSC::Unknown>>::value);
+#if USE(JSVALUE64)
+    // We can memset only in JSVALUE64 since empty value is zero. On the other hand, JSVALUE32_64's empty value is not zero.
```

Comment:
Comment is great but code should be `#if !USE(JSVALUE64)` and only include setting to `false`, rely on inheritance for the true case.

Reply:
Right, fixed.